feed/feed.go

```diff
@@ -225,15 +239,75 @@ func ServiceWithDefaultFeeds(pgxStore *store.PGXStore) *Service {
 	r.Register(Meta{ID: "furry-hot"}, scoreBasedGenerator(1.85, time.Hour*2))
 
 	// Reverse chronological based feeds
-	r.Register(Meta{ID: "furry-new"}, chronologicalGenerator([]string{}, []string{}))
-	r.Register(Meta{ID: "furry-fursuit"}, chronologicalGenerator([]string{bff.TagFursuitMedia}, []string{}))
-	r.Register(Meta{ID: "fursuit-nsfw"}, chronologicalGenerator([]string{bff.TagFursuitMedia, bff.TagNSFW}, []string{}))
-	r.Register(Meta{ID: "fursuit-clean"}, chronologicalGenerator([]string{bff.TagFursuitMedia}, []string{bff.TagNSFW}))
-	r.Register(Meta{ID: "furry-art"}, chronologicalGenerator([]string{bff.TagArt}, []string{}))
-	r.Register(Meta{ID: "art-clean"}, chronologicalGenerator([]string{bff.TagArt}, []string{bff.TagNSFW}))
-	r.Register(Meta{ID: "art-nsfw"}, chronologicalGenerator([]string{bff.TagNSFW, bff.TagArt}, []string{}))
-	r.Register(Meta{ID: "furry-nsfw"}, chronologicalGenerator([]string{bff.TagNSFW}, []string{}))
-	r.Register(Meta{ID: "furry-comms"}, chronologicalGenerator([]string{bff.TagCommissionsOpen}, []string{}))
+	r.Register(Meta{ID: "furry-new"}, chronologicalGenerator(chronologicalGeneratorOpts{
+		RequireTags: []string{},
+		ExcludeTags: []string{},
+
+		RequireHashtags: []string{},
+		ExcludeHashtags: []string{},
+	}))
+	r.Register(Meta{ID: "furry-fursuit"}, chronologicalGenerator(chronologicalGeneratorOpts{
+		RequireTags: []string{bff.TagFursuitMedia},
+		ExcludeTags: []string{},
+
+		RequireHashtags: []string{"fursuitfriday", "fursuit", "murrsuit", "mursuit"},
+		ExcludeHashtags: []string{},
+		MustHaveMedia:   true,
+	}))
+	r.Register(Meta{ID: "fursuit-nsfw"}, chronologicalGenerator(chronologicalGeneratorOpts{
+		RequireTags: []string{bff.TagFursuitMedia, bff.TagNSFW},
+		ExcludeTags: []string{},
+
+		RequireHashtags: []string{"fursuitfriday", "fursuit", "murrsuit", "mursuit", "nsfw"},
+		ExcludeHashtags: []string{},
+		MustHaveMedia:   true,
+	}))
+	r.Register(Meta{ID: "fursuit-clean"}, chronologicalGenerator(chronologicalGeneratorOpts{
+		RequireTags: []string{bff.TagFursuitMedia},
+		ExcludeTags: []string{bff.TagNSFW},
+
+		RequireHashtags: []string{"fursuitfriday", "fursuit"},
```

Comment:
At least with the way the query is currently written I don't think this behaves as you expect it to :'). The "exclude" uses && which checks for any elements in common (e.g if a single tag matches, exclude the posts) but the "require" uses @> which checks that all values are contained within the hashtag on the post. So here, you'd only get posts including fursuitfriday AND fursuit.

The only time we currently actually need to filter by posts containing at least two hashtags is for the NSFW feeds, which will change when post self-tagging is adding to bluesky. I think maybe, we need to introduce another column `is_nsfw` and then change this from `RequireHashtags` to `IncludeHashtags` and match the overlap behaviour of `ExcludeHashtags` (e.g any matches).